cmd/deploy.go

```diff
@@ -111,22 +111,32 @@ func deployFn(_ *cobra.Command, _ []string) error {
 
 	opts := []clab.ClabOption{
 		clab.WithTimeout(timeout),
-		clab.WithTopoFile(topo, varsFile),
-		clab.WithNodeFilter(nodeFilter),
 		clab.WithRuntime(rt,
 			&runtime.RuntimeConfig{
 				Debug:            debug,
 				Timeout:          timeout,
 				GracefulShutdown: graceful,
 			},
 		),
+		clab.WithTopoFile(topo, varsFile),
+		clab.WithNodeFilter(nodeFilter),
 		clab.WithDebug(debug),
 	}
```

Comment:
@steiler do you recall why you had to move WithTopoFile and WithNodeFilter after WithRuntime?

Reply:
Yes, it has to do with attaching interfaces to the `mgmt-net`. The mgmt-net -> bridge is resolved via the runtime.
The FakeMgmtNetNode is created as a singleton and thereby initialized only on the first call. So the bridge infos need to be present before any resolution takes place.
I can take a look at if it structurally makes sense to delay the population of the fakemgmtnetbridge with the bridges name.

Reply:
The bigger issue even was the CheckBridgeExists validation check that ran quite early... anyways I've prepared a change for this.
Take a look at https://github.com/steiler/containerlab/pull/2 that PR is based on this branch.